pkg/segment/query/metricsquery.go

```diff
@@ -355,14 +356,16 @@ func applyMetricsOperatorOnSegments(mQuery *structs.MetricsQuery, allSearchReqes
 	// for each metrics segment, apply a single metrics segment search
 	// var tsidInfo *tsidtracker.AllMatchedTSIDs
 
+	// allocate 1 MB buffer
```

Comment:
I think starting with 50 KB is good enought, we anyways have code to extend it during runtime